### PR TITLE
Get rid of blank role attribute and rowgroup role on the innerGrid

### PR DESCRIPTION
**Pull request by niik**
This was incorrectly removed in https://github.com/desktop/desktop/commit/8562094757150be508ce07cf0847b746fcc07872. There's a difference between having a blank role attribute and not having the attribute at all (which is what we want here). Additionally we don't want the inner grid to have role `rowgroup` because we're using role listbox on the containing div.

### CODE REVIEW

**Review by tidy-dev — DISMISSED** [review]
✨  Thank you!

**Comment by tidy-dev** [comment]
Note.. just noticed that this causes console errors.
image

**Comment by niik** [comment]
> Note.. just noticed that this causes console errors.

Welp, this turned this small change into a bigger one. I ended up moving all the a11y props and the keyboard event handler down to the inner grid instead of the container div. I also added a hack to unset the default props and prop requirements of the Grid component because they don't make a lot of sense in our case.

File: app/src/ui/index.tsx [hunk 1]
```diff
@@ -352,6 +353,26 @@ ipcRenderer.on('url-action', (_, action) =>
...
+// > state.
+//
+// 1. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label
+// 2. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-readonly
+;(function (defaults: Record<string, unknown>, types: Record<string, unknown>) {
+  ;['aria-label', 'aria-readonly'].forEach(k => {
+    delete defaults[k]
+    delete types[k]
+  })
+})(Grid.defaultProps, Grid.propTypes)
```
Comment (tidy-dev) [inline app/src/ui/index.tsx:374]: Meh.. I was wondering if we were going to need to "manually" remove these props.

**Review by tidy-dev — APPROVED** [review]
✨ Works great!

This definitely resolves, [#3245](https://github.com/github/accessibility-audits/issues/3245) - announcing list item positioning.

And I believe it fixed [#3261](https://github.com/github/accessibility-audits/issues/3261) as well. When I tested that last one on windows, I got "list clickable" not "out of grouping list clickable" as described in the ticket. With this fix, I got "list". (On mac, it was "listbox", now it is "empty, listbox")

**Pull request merged**